### PR TITLE
Performance and product size improvements

### PR DESCRIPTION
**Pull request by jrs65**

### CODE REVIEW

File: drift/core/telescope.py [hunk 1]
```diff
@@ -175,6 +182,18 @@ class TransitTelescope(config.Reader, ctime.Observer, metaclass=abc.ABCMeta):
...
+    skip_freq : list
+        Frequency indices (with the set of frequencies defined by the other parameters)
+        to skip. Skipped frequencies are considered to be present, *but* their beam
+        transfer matrices are implicitly zero and thus are skipped in the beam transfer
+        matrix calculation. This is useful for RFI channels that we know will be masked.
+    skip_baselines : list
+        Baseline indices to skip. Like skipped frequencies, skipped baselines are
+        considered to be present, *but* their beam transfer matrices are implicitly zero
+        and thus are skipped in the beam transfer matrix calculation.
+    beam_cache_size : float
```
Comment (sjforeman) [inline drift/core/telescope.py:194]: Specify default value in docstring

File: drift/core/beamtransfer.py [hunk 2]
```diff
@@ -140,12 +157,42 @@ class BeamTransfer(object):
     telescope : drift.core.telescope.TransitTelescope, optional
         Telescope object to use for calculation. If `None` (default), try to
         load a cached version from the given directory.
+
+    Attributes
+    ----------
+    mem_chunk : float
```
Comment (sjforeman) [inline drift/core/beamtransfer.py:163]: Specify default values for all these attributes in docstring

**Review by sjforeman — APPROVED** [review]

File: drift/util/_fast_tools.pyx [hunk 3]
```diff
@@ -0,0 +1,406 @@
...
+from libc.stdlib cimport abort, malloc, free
+from libc.math cimport sin, cos, tan, exp, hypot, M_PI, M_PI_2, M_LN2
+
+ctypedef double complex complex128
+
+cdef extern from "complex.h" nogil:
+    complex128 conj(complex128)
+
+
+def sph_to_cart(sph_coords):
```
[the remark this answers is not in the document]
Comment (jrs65) [reply drift/util/_fast_tools.pyx:16]: When I started writing them it was a combination of convenience, and the fact that the fast versions had a slightly restricted API (i.e. only 2D float64 arrays were accepted). Now I've relaxed those I can move these out as suggested.

File: drift/util/_fast_tools.pyx [hunk 4]
```diff
@@ -0,0 +1,406 @@
...
+        The amplitude beam at each requested angle.
+    """
+    cdef double tan2, alpha
+    cdef double[::1] et_view
+    cdef Py_ssize_t i, n
+
+    et = np.empty_like(sintheta)
+    et_view = et
+
+    alpha = M_LN2 / (2 * tan(fwhm / 2.0) ** 2)
```
[the remark this answers is not in the document]
Comment (jrs65) [reply drift/util/_fast_tools.pyx:399]: Good point. I'd totally forgotten about that issue. I *think* there's a typo in the paper itself, but I'll go through and double check.

**Comment by jrs65** [comment]
@sjforeman I think I've addressed all your comments, but I can imagine this branch might not pass the CI until the cora changes have been merged.

**Pull request merged**